cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java

```diff
@@ -324,12 +326,14 @@ public ArtifactDetail addArtifact(Id.Artifact artifactId, File artifactFile,
    *                        If null, the given artifact does not extend another artifact
    * @param additionalPlugins the set of additional plugin classes to add to the plugins found through inspection.
    *                          If null, no additional plugin classes will be added
+   * @param properties properties for the artifact
    * @throws IOException if there was an exception reading from the artifact store
    * @throws ArtifactRangeNotFoundException if none of the parent artifacts could be found
    */
   public ArtifactDetail addArtifact(Id.Artifact artifactId, File artifactFile,
                                     @Nullable Set<ArtifactRange> parentArtifacts,
-                                    @Nullable Set<PluginClass> additionalPlugins)
+                                    @Nullable Set<PluginClass> additionalPlugins,
+                                    @Nullable Map<String, String> properties)
```

Comment:
instead of ``@Nullable``, we can have an addArtifact method without properties, that can call this method with an empty map.